agent/pkg/models/models.go

```diff
@@ -194,13 +198,22 @@ type ExtendedCreator struct {
 }
 
 type FullEntryWithPolicy struct {
-	RulesMatched []shared.RulesMatched `json:"rulesMatched,omitempty"`
-	Entry        har.Entry             `json:"entry"`
-	Service      string                `json:"service"`
+	RulesMatched []rules.RulesMatched `json:"rulesMatched,omitempty"`
+	Entry        har.Entry            `json:"entry"`
+	Service      string               `json:"service"`
+}
+
+func (fewp *FullEntryWithPolicy) UnmarshalData(entry MizuEntry, harEntry har.Entry) {
```

Comment:
Implement `DataUnmarshaler` interface.
See for example how `FullEntryDetails` implements and uses `UnmarshalData`.